src/main/kotlin/org/rust/ide/inspections/lints/RsDeprecationInspection.kt

```diff
@@ -43,24 +47,41 @@ class RsDeprecationInspection : RsLintInspection() {
     private fun checkAndRegisterAsDeprecated(identifier: PsiElement, original: PsiElement, holder: RsProblemsHolder) {
         if (original is RsOuterAttributeOwner) {
             val attr = original.queryAttributes.deprecatedAttribute ?: return
-            holder.registerLintProblem(identifier, attr.extractDeprecatedMessage(identifier.text))
+            val (message, highlightType) = attr.extractDeprecatedMessage(identifier.text)
+            holder.registerProblem(identifier, message, highlightType)
```

Comment:
Note, `registerLintProblem` takes into account the current [lint level](https://doc.rust-lang.org/rustc/lints/levels.html) of the corresponding lint but new implementation doesn't do it, as I can see

Reply:
Ah, I was wondering if there's any problem with not using `registerLintProblem`. The current lint API is slightly cumbersome if the lint has several highlight types for various situations.

It would be nice to have the option to send a custom object to `registerLintProblem` that would dynamically allow selecting the highlight type. I tried to implement it, let me know what you think.

Reply:
Interesting that rustc shows the same warnings regardless of the version. Maybe we shouldn't use different highlighting too

Reply:
Another option is introducing another lint with the same `id`. I'm not sure whether it is ok or not

Reply:
> Another option is introducing another lint with the same id. I'm not sure whether it is ok or not

That sounds.. a bit confusing :)

> Interesting that rustc shows the same warnings regardless of the version. Maybe we shouldn't use different highlighting too.

I think that we can do better than rustc here. I would also show a different highlight type, since sometimes you might not care about future versions if your version is fixed, so it might be annoying. If it's not deprecated currently, it should be OK to use, so weak warning seems right to me.